src/snowflake/snowpark/_internal/udf_utils.py

```diff
@@ -446,7 +478,7 @@ def extract_return_input_types(
             return_type_from_type_hints, PandasDataFrameType
         ):  # vectorized UDTF
             return_type = PandasDataFrameType(
-                [x.datatype for x in return_type], [x.name for x in return_type]
+                [x.datatype for x in return_type], [x.name for x in return_type]  # type: ignore
```

Comment:
There's some sort of coupling between the `return_type_from_type_hints` and `return_type` variables that mypy isn't able to understand. Fixing would require a larger refactor than I think is appropriate for this PR.